src/Serializers/RestSerializer.test.ts

```diff
@@ -72,4 +73,54 @@ describe("RestSerializer", () => {
 
   });
 
+  describe("serializeQueryParams", () => {
+
+    it("convert the given IQueryParams object into a url-encoded string", async () => {
+
+      const fakeQueryParamData = {
+        page: random.number().toString(),
+        pageSize: random.number().toString(),
+        grade: random.number().toString(),
+        current: random.boolean().toString(),
+        values: [name.firstName(), name.firstName()],
+        sort: [
+          {key: name.firstName(), direction: "asc" as SortDirection},
+          {key: name.firstName(), direction: "desc" as SortDirection},
+        ],
+      };
+
+      const restSerializer = new RestSerializer(FakeModel);
+      const urlEncodedString = await restSerializer.serializeQueryParams(fakeQueryParamData);
+     
+      // tslint:disable-next-line:max-line-length (with the backslash \ line continuation it seems to put in an extra space)
+      expect(urlEncodedString).to.equal(`page=${fakeQueryParamData.page}&pageSize=${fakeQueryParamData.pageSize}&grade=${fakeQueryParamData.grade}&current=${fakeQueryParamData.current}&values=${fakeQueryParamData.values[0]},${fakeQueryParamData.values[1]}&sort=${fakeQueryParamData.sort[0].key},${fakeQueryParamData.sort[1].key}${encodeURIComponent(":")}desc`);
+
+    });
+
+    it("convert the given IQueryParams object into a url-encoded string with special characters in names", async () => {
+      
+      const name1 = `${name.firstName()}%`;
+      const name2 =  `${name.firstName()},o`;
+
+      const fakeQueryParamData = {
+        page: random.number(),
+        pageSize: random.number(),
+        grade: random.number(),
+        current: random.boolean(),
+        values: [name.firstName(), name.firstName()],
+        sort: [
+          {key: `${name1}`, direction: "asc" as SortDirection},
+          {key: `${name2}`, direction: "desc" as SortDirection},
+        ],
+      };
+
+      const restSerializer = new RestSerializer(FakeModel);
+      const urlEncodedString = await restSerializer.serializeQueryParams(fakeQueryParamData);
+
+      // tslint:disable-next-line:max-line-length (with the backslash \ line continuation it seems to put in an extra space)
```

Comment:
@jmadson  This was my suggestion rather than breaking this up via `+` or `\`, the latter of which was causing whitespace discrepancies between the actual and expected strings.

Reply:
no worries! 👍